mava/components/tf/modules/exploration/exploration_scheduling.py

```diff
@@ -237,14 +237,89 @@ def decrement_epsilon(self) -> float:
         """
         return self._epsilon
 
+
 def apex_exploration_scheduler(
-        num_executors: int = 1,
-        epsilon: float = 0.4,
-        alpha: float = 7.,
-    ) -> Mapping[str, ConstantScheduler]:
+    num_executors: int = 1,
+    epsilon: float = 0.4,
+    alpha: float = 7.0,
+) -> Mapping[str, ConstantScheduler]:
+    """Returns a scheduler with a single espilon per executor.
+
+    The espilons are given by Ape-X formula
+
+    Returns:
+        Ape-X exploration scheduler exploration_scheduler_fn
+    """
     exploration_scheduler_fn: Dict = {}
     for executor_id in range(num_executors):
         executor = f"executor_{executor_id}"
-        epsilon_i = epsilon**(1 + alpha * executor_id / (num_executors-1))
+        epsilon_i = epsilon ** (1 + alpha * executor_id / (num_executors - 1))
         exploration_scheduler_fn[executor] = ConstantScheduler(epsilon=epsilon_i)
-    return exploration_scheduler_fn
\ No newline at end of file
+    print("exploration_scheduler_fn", exploration_scheduler_fn)
+    return exploration_scheduler_fn
+
+
+def monotonic_ma_apex_exploration_scheduler(
+    agent_ids: List[str],
+    num_executors: int = 1,
+    epsilon: float = 0.4,
+    alpha: float = 7.0,
+) -> Mapping[str, Mapping[str, ConstantScheduler]]:
+    """Returns a scheduler with distinct espilons per executor and agent.
+
+    The schedulers are assigned in increasing order to all executor.
+    Within every scheduler, the espilons are randomy assigned to agents.
+
+    Returns:
+        Monotonic Multi-Agent Ape-X exploration scheduler exploration_scheduler_fn
+
+    """
+    num_agents = len(agent_ids)
+    exploration_scheduler_fn: Dict = {}
```

Comment:
This might be a good place to use a `defaultdict(dict)`

Reply:
Done.